lib/3scale/backend/worker_async.rb

```diff
@@ -73,9 +73,10 @@ def process_all
           break if @shutdown
 
           semaphore.async { perform(job) }
+          barrier.wait if barrier.size > semaphore.limit
         end
       ensure
-        barrier.wait
+        barrier.stop
```

Comment:
I think we previously discussed that we don't want `stop`because it cancels tasks and that would lead to missed tasks as they are already out of Redis. No?

https://www.rubydoc.info/gems/async/2.2.1/Async/Barrier#stop-instance_method

Reply:
Right, I removed the calls to `stop` (https://github.com/3scale/apisonator/pull/376/commits/82de3b1ed07ec26e02e02f24abb921415d63a89a).